corehq/apps/reports/standard/deployments.py

```diff
@@ -155,14 +155,16 @@ def get_sorting_block(self):
                     sort_dict = {
                         sort_prop: {
                             "order": sort_dir,
-                            "nested_path": "reporting_metadata.last_submissions",
                             "nested_filter": {
                                 "term": {
                                     self.sort_filter: self.selected_app_id
                                 }
                             }
                         }
                     }
+                    if sort_prop != 'username.exact':
+                        # we can't specify a nested_path when sorting by username
+                        sort_dict[sort_prop]['nested_path'] = "reporting_metadata.last_submissions"
```

Comment:
I'm trying to wrap my head around why `"reporting_metadata.last_submission"` is always the right `nested_path`, except for `username.exact`, and I think it's because `username.exact` doesn't start with `reporting_metadata.last_submission`, unlike, say, `reporting_metadata.last_submissions.submission_date`, which does. Looking at https://github.com/dimagi/commcare-hq/blob/86ba3862f89b8d5eb6c41913cb23e774429faac9/corehq/apps/reports/standard/deployments.py#L71-L94, which I think is the list of columns in this report including which ES properties they map to, I see some others that don't start with `reporting_metadata.last_submissions.submission_date` like for example `reporting_metadata.last_syncs.sync_date`.

Looking at https://www.elastic.co/guide/en/elasticsearch/reference/1.7/search-request-sort.html versus https://www.elastic.co/guide/en/elasticsearch/reference/2.4/search-request-sort.html, the relevant part that changed seems to be

1.7
> Because offer is the closest inherited nested field, it is picked as nested_path.
2.4
> The nested_path needs to be specified; otherwise, elasticsearch doesn’t know on what nested level sort values need to be captured.

That would lead me to believe that if we want to preserve the 1.7 default behavior, then we should always be picking `nested_path = '.'.join(sort_prop.split('.')[:-1])`, but honestly I don't understand what nested_path does, and how it would be different to pick, say, `"reporting_metadata"` instead of `"reporting_metadata.last_submission"` as the `nested_path`, and what it means / what the behavior is if `sort_prop` is not a descendant of `nested_path` (as is currently the case for `username.exact`, `reporting_metadata.last_syncs.sync_date`, etc.

This is the only other thing that page says about `nested_path`
> Defines on which nested object to sort. The actual sort field must be a direct field inside this nested object. When sorting by nested field, this field is mandatory.
which doesn't make things much clearer to me. Do you get what it means?

Finally the `exact` in `username.exact` might be treated differently somehow than nested properties, maybe that's why it seems to want you to leave off the `nested_property`. So all that said right now, trying to piece these things together, the following seems like the right change here:

```suggestion
                    sort_prop_path = sort_prop.split('.')
                    if sort_prop_path[-1] == 'exact':
                        sort_prop_path.pop()
                    sort_prop_path.pop()
                    if sort_prop:
                        sort_dict[sort_prop]['nested_path'] = '.'.join(sort_prop_path)
```